### PR TITLE
add manifest file to extension-ui

### PR DESCRIPTION
**Pull request by est8906**
it is excelent

### CODE REVIEW

**Review by Ruth708090 — APPROVED** [review]
ב"ה הקוד נראה מעולה קריא ונהיר!

File: extension-ui/background.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    if (blockedSites.some(site => hostname.includes(site))) {
+      chrome.tabs.get(details.tabId, (tab) => {
+        if (tab.url.startsWith('chrome://') || tab.url.startsWith('about:')) {
+          return;
+        }
+        chrome.scripting.executeScript({
+          target: { tabId: details.tabId },
+          func: () => {
+            window.stop();
+            window.location.href = chrome.runtime.getURL('hoops.html');
```
Comment (EA171) [inline extension-ui/background.js:24]: please add TODO comment - to remember add UI for the oops window

File: extension-ui/content.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+chrome.storage.local.get("blockedSites", (data) => {
+    const blockedSites = data.blockedSites || [];
+    console.log({blockedSites});
```
Comment (EA171) [inline extension-ui/content.js:3]: remove console.log

File: extension-ui/hoops.html [hunk 3]
```diff
@@ -0,0 +1,30 @@
+<!DOCTYPE html>
+<html>
+
+<head>
+  <title>Access Blocked</title>
+  <style>
```
Comment (EA171) [inline extension-ui/hoops.html:6]: please move the styles to separate file

File: extension-ui/popup.html [hunk 4]
```diff
@@ -0,0 +1,47 @@
+<!DOCTYPE html>
+<html lang="en">
+
+<head>
+  <meta charset="UTF-8">
+  <title>Website Blocker</title>
+  <style>
+    body {
```
Comment (EA171) [inline extension-ui/popup.html:8]: please move styled to separate window

File: extension-ui/popup.html [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    }
+
+    .hidden {
+      display: none;
+    }
+  </style>
+</head>
+
+<body>
+  <div class="container">
```
Comment (EA171) [inline extension-ui/popup.html:28]: please add TODO comment - to use generic components and style this page

File: extension-ui/popup.js [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+  const blockForm = document.getElementById("blockForm");
+  const siteInput = document.getElementById("siteInput");
+
+  blockForm.addEventListener("submit", (event) => {
+    event.preventDefault();
+    const inputUrl = siteInput.value.trim();
+    if(inputUrl){
+      url = new URL(inputUrl)
+    // const url = new URL(inputUrl);
```
Comment (EA171) [inline extension-ui/popup.js:36]: remove comment

**Review by EA171 — CHANGES_REQUESTED** [review]

File: extension-ui/background.js [hunk 7]
```diff
@@ -0,0 +1,32 @@
+chrome.runtime.onInstalled.addListener(() => {
+  chrome.storage.local.set({ blockedSites: [] });
+});
+
+chrome.webNavigation.onBeforeNavigate.addListener((details) => {
+  const url = new URL(details.url);
```
Comment (SariGlick) [inline extension-ui/background.js:6]: put this in try-catch for invalid URLs

File: extension-ui/background.js [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+  if (url.protocol === 'chrome:' || url.protocol === 'about:') {
+    return;
+  }
+ 
+  chrome.storage.local.get(["blockedSites"], (data) => {
```
Comment (SariGlick) [inline extension-ui/background.js:11]: wherever you used chrome.storage.local.get, please save in to variable and use it. only if it's empty, get the data again from the local storage

**Review by SariGlick — CHANGES_REQUESTED** [review]

**Review by e171271 — APPROVED** [review]
opened another sub-task for adding support for scss files

**Review by EA171 — APPROVED** [review]

**Review by SariGlick — APPROVED** [review]

**Pull request merged**